development/tools/pkg/rendertemplates/genericcomponent.go

```diff
@@ -13,8 +13,8 @@ func (j *Job) changeExtraRefsBase(base string) {
 	}
 }
 
-// GenerateComponentJobs generates jobs for components
-func (r *RenderConfig) GenerateComponentJobs(global map[string]interface{}) {
+// AppendJobs generates jobs for components if necessary and appends pre/post/main jobs to the list if they exist
```

Comment:
To which list we append this jobs? Is it a list for rendering files from templates or for some another processing step?
I think "if they exist" is not needed as the function generates jobs if needed.
Add some two words about accepted parameter. What global is? Is it map of global configSets?